src/java.base/unix/classes/sun/nio/fs/UnixFileAttributeViews.java

```diff
@@ -72,38 +72,23 @@ public void setTimes(FileTime lastModifiedTime,
             // permission check
             file.checkWrite();
 
+            // use a file descriptor if possible to avoid a race due to
+            // accessing a path more than once as the file at that path could
+            // change.
+            // if path is a symlink, then the open should fail with ELOOP and
+            // the path will be used instead of the file descriptor.
             boolean haveFd = false;
-            boolean useFutimes = false;
-            boolean useFutimens = false;
-            boolean useLutimes = false;
-            boolean useUtimensat = false;
             int fd = -1;
             try {
-                if (!followLinks) {
-                    // these path-based syscalls also work if following links
-                    if (!(useUtimensat = utimensatSupported())) {
-                        useLutimes = lutimesSupported();
-                    }
-                }
-                if (!useUtimensat && !useLutimes) {
-                    fd = file.openForAttributeAccess(followLinks);
-                    if (fd != -1) {
-                        haveFd = true;
-                        if (!(useFutimens = futimensSupported())) {
-                            useFutimes = futimesSupported();
-                        }
-                    }
-                }
+                fd = file.openForAttributeAccess(followLinks);
```

Comment:
This throws if O_NOFOLLOW is not supported. Are there any platforms where this is possible?

Reply:
I believe some old AIX versions. But they are no longer supported. We require 7.2.